recipes-connectivity/networkmanager/networkmanager-1.4.4/update-resolv-conf.patch

```diff
@@ -0,0 +1,176 @@
+#OpenXT:Update resolv.conf for different interfaces as dnsmasq script running on different bridges makes use of it.
+#       resolv.conf for different interfaces is done by appending the interface name to 'resolv.conf.' file.
+#       For example: resolv.conf for interface wlan0 would be resolv.conf.wlan0
+--- a/src/dns-manager/nm-dns-manager.c
++++ b/src/dns-manager/nm-dns-manager.c
+@@ -147,6 +147,10 @@ struct _NMDnsManagerClass {
+ 	GObjectClass parent;
+ };
+ 
++#ifndef RESOLV_CONF
++#define RESOLV_CONF "/var/volatile/etc/resolv.conf"
++#endif
++
+ G_DEFINE_TYPE (NMDnsManager, nm_dns_manager, G_TYPE_OBJECT)
+ 
+ NM_DEFINE_SINGLETON_INSTANCE (NMDnsManager);
+@@ -651,6 +655,7 @@ dispatch_resolvconf (NMDnsManager *self,
+ 
+ static SpawnResult
+ update_resolv_conf (NMDnsManager *self,
++                    const char *resolv_path,
+                     char **searches,
+                     char **nameservers,
+                     char **options,
+@@ -663,7 +668,7 @@ update_resolv_conf (NMDnsManager *self,
+ 	gs_free char *content = NULL;
+ 	SpawnResult write_file_result = SR_SUCCESS;
+ 	int errsv;
+-	const char *rc_path = _PATH_RESCONF;
++	const char *rc_path = resolv_path;
+ 	nm_auto_free char *rc_path_real = NULL;
+ 
+ 	/* If we are not managing /etc/resolv.conf and it points to
+@@ -834,7 +839,7 @@ update_resolv_conf (NMDnsManager *self,
+ 		return SR_ERROR;
+ 	}
+ 
+-	if (symlink (MY_RESOLV_CONF, RESOLV_CONF_TMP) == -1) {
++	if (symlink (RESOLV_CONF, RESOLV_CONF_TMP) == -1) {
```

Comment:
Yes, this is never reached.
Yes, `rc_manager` is set to `NM_DNS_MANAGER_RESOLV_CONF_MAN_FILE`, because we are writing to `/var/volatile/etc/resolv.conf.XXX` (XXX being the name of an interface), which is a regular file.

Reply:
But I made this change thinking if `rc_manager` is ever set to `NM_DNS_MANAGER_RESOLV_CONF_SYMLINK`. I can remove this if you're thinking this will be redundant.